state/cockroachdb/cockroachdb_access.go

```diff
@@ -310,66 +405,68 @@ func (p *cockroachDBAccess) BulkDelete(ctx context.Context, req []state.DeleteRe
 	return err
 }
 
-func (p *cockroachDBAccess) ExecuteMulti(ctx context.Context, request *state.TransactionalStateRequest) error {
-	p.logger.Debug("Executing PostgreSQL transaction")
+func (c *CockroachDBAccess) ExecuteMulti(ctx context.Context, request *state.TransactionalStateRequest) error {
+	c.logger.Debug("Executing CockroachDB transaction")
 
-	tx, err := p.db.Begin()
+	tx, err := c.db.Begin()
 	if err != nil {
 		return err
 	}
 
-	for _, o := range request.Operations {
-		switch o.Operation {
-		case state.Upsert:
-			var setReq state.SetRequest
-
-			setReq, err = getSet(o)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-			err = p.Set(ctx, &setReq)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-		case state.Delete:
-			var delReq state.DeleteRequest
-
-			delReq, err = getDelete(o)
-			if err != nil {
-				tx.Rollback()
-				return err
-			}
-
-			err = p.Delete(ctx, &delReq)
-			if err != nil {
-				tx.Rollback()
-				return err
+	performTx := func() error {
```

Comment:
I would really like to avoid using lambdas when it's just to be able to `defer` a call inside, because they are not light on the heap. Was there an issue with the previous code?

Reply:
Done